app/src/main/java/org/oppia/app/topic/play/StorySummaryAdapter.kt

```diff
@@ -109,27 +113,81 @@ class StorySummaryAdapter(
 
       val chapterList = storySummaryViewModel.storySummary.chapterList
       binding.chapterRecyclerView.adapter = ChapterSummaryAdapter(chapterList, chapterSummarySelector)
+      if (currentExpandedChapterListIndex != null && currentExpandedChapterListIndex == position) {
+        val rotate = AnimationUtils.loadAnimation(context, R.anim.rotation_clockwise_180)
+        binding.chapterListDropDownIcon.startAnimation(rotate)
+        expand(binding.chapterListContainer)
+      } else {
+        val rotate = AnimationUtils.loadAnimation(context, R.anim.rotation_anti_clockwise_180)
+        binding.chapterListDropDownIcon.startAnimation(rotate)
+        collapse(binding.chapterListContainer)
+      }
 
       binding.root.setOnClickListener {
-        val previousIndex: Int? = currentExpandedChapterListIndex
-        currentExpandedChapterListIndex =
-          if (currentExpandedChapterListIndex != null && currentExpandedChapterListIndex == position) {
-            null
-          } else {
-            position
-          }
-        expandedChapterListIndexListener.onExpandListIconClicked(currentExpandedChapterListIndex)
-        if (previousIndex != null && currentExpandedChapterListIndex != null && previousIndex == currentExpandedChapterListIndex) {
-          notifyItemChanged(currentExpandedChapterListIndex!!)
+        val previousItem = currentExpandedChapterListIndex
+        if (binding.chapterListContainer.isVisible) {
+          currentExpandedChapterListIndex = null
         } else {
-          if (previousIndex != null) {
-            notifyItemChanged(previousIndex)
+          currentExpandedChapterListIndex = position
+        }
+        expandedChapterListIndexListener.onExpandListIconClicked(currentExpandedChapterListIndex)
+        notifyItemChanged(position)
+        if (previousItem != null && previousItem != position) {
```

Comment:
This logic isn't trivial to follow. Can we add tests to make sure that the expand/collapse works as expected & stays working? We shouldn't test the animations, just that the state changes are being reflected correctly.

Reply:
I have updated all test cases:

Check the following tests which checks this functionality:
`testTopicPlayFragment_loadFragmentWithTopicTestId0_clickExpandListIconIndex0_clickExpandListIconIndex1_chapterListForIndex0IsNotDisplayed`
`testTopicPlayFragment_loadFragmentWithTopicTestId0_clickExpandListIconIndex1_clickExpandListIconIndex0_chapterListForIndex0IsNotDisplayed`